### ISSUE TITLE
Company Manager cannot view user profile after clearing all courses from a User

### ISSUE THREAD

**Issue by dallenk** [iomad/iomad#1769]
tested on the sandbox and easily duplicated.
as a company manager, create a user and assign some courses, the users profile can be viewed by the company manager.
in the "User Report" area, clear course and purge records for the user.
(php warning also shows:
`Notice: Constant CERTIFICATE already defined in /var/www/vhosts/sandbox.domain.ca/iomad/local/iomad_track/classes/observer.php on line 29`

user profile isn't available anymore from Dashboard / Users / <username>

![image](https://user-images.githubusercontent.com/2725066/222619969-beeb9f50-34bb-47b8-ba54-a71d4163b3bc.png)

**Comment by turf212** [comment]
This is Moodle throwing the capability check.  If the user is on the same course as the manager then there is one route.  If they aren't (e.g. you are using educators instead of managers or only licensed courses) then it checks a different route.

I'll add a check to make sure we are respecting this as a check to see if the link should be provided or not.

**Comment by dallenk** [comment]
I see the commit on 400_STABLE but not 401.  does this also apply to 401?  I manually added the code to my install at 401_stable and the error remained.

edit2:   I noticed the permissions **[course]moodle/user:viewdetails** and **[user]moodle/user:viewalldetails** are checked in the commit.  I actually tried this option before creating the issue (sorry, I should have mentioned that)  enabling either of these options appear to be global and not restricted by company at all, so all I have to do as a company manager is change profile.php?id=xxx  and I can see the profile of any user, in any company (but maybe only if they are enrolled in the course).

In my setup, a company can share a course as open or closed, a user of any other company can self-enroll into that open-shared course. 
Other courses are closed shared and only assigned to some of the companies, which their users can also self-enroll into. 

Perhaps this is a scenario IOMAD wasn't designed for?
